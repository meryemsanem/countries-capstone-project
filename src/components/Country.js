import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './styles/Country.css';

function Country({ country }) {
  const { name, flag } = country;
  return (
    <>
      <div className="card">
        <NavLink
          to={`/details/${name}`}
          className="details-link"
          data-testid="details-link"
        >
          <div className="image">
            <img src={flag} alt="country" />
          </div>
          <h1 className="country-name">{name}</h1>
        </NavLink>
      </div>
    </>
  );
}

export default Country;

Country.propTypes = {
  country: PropTypes.shape({
    name: PropTypes.string.isRequired,
    flag: PropTypes.string.isRequired,
  }).isRequired,
};
