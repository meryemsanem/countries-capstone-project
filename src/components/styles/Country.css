.country-name {
  text-align: center;
  font-size: 20px;
  color: white;
  text-transform: uppercase;
}

img {
  width: 60%;
}

.card {
  justify-content: center;
  display: flex;
  text-align: center;
}

.details-link {
  text-decoration: none;
}
