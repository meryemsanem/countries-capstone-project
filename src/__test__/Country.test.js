import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import Country from '../components/Country';

describe('Country Component', () => {
  const mockCountry = {
    name: 'Test Country',
    flag: 'test-flag-url',
  };

  it('renders correctly', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Country country={mockCountry} />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
