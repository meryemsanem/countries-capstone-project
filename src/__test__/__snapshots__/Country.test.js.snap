// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <a
      class="details-link"
      data-testid="details-link"
      href="/details/Test Country"
    >
      <div
        class="image"
      >
        <img
          alt="country"
          src="test-flag-url"
        />
      </div>
      <h1
        class="country-name"
      >
        Test Country
      </h1>
    </a>
  </div>
</DocumentFragment>
`;
